src/server/DeviceServer.js

```diff
@@ -358,7 +358,13 @@ class DeviceServer {
         await this._deviceAttributeRepository.getById(deviceID);
 
       if (!deviceAttributes || !deviceAttributes.ownerID) {
-        device.sendReply('SubscribeFail', message.getId());
+        // not sure if sending 'ok subscribe reply' right in this case, but with
+        // SubscribeFail the device reconnects to the cloud infinitely
+        device.sendReply('SubscribeAck', message.getId());
+        logger.log(
+          `device with ID ${deviceID} wasn't subscribed to` +
+        `${messageName} MY_DEVICES event: the device is unclaimed.`
```

Comment:
This still isn't 100% correct. You're missing a tab on line 366 and a comma at the end.  I'm going to merge but fix this next time you're looking at this code.